### PR TITLE
ON-12431 # Upgraded Joi

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: src/lib/forms-schema/common.ts [hunk 1]
```diff
@@ -43,39 +43,48 @@ export const SEARCH_OPTION_TYPE = 'SEARCH'
 export const optionTypes = [CUSTOM_OPTION_TYPE, DYNAMIC_OPTION_TYPE]
 
 export const JoiRange = Joi.extend((joi: typeof Joi) => ({
+  type: 'range',
   base: joi.number(),
-  name: 'range',
-  language: {
-    within: 'Must not exceed range of values {{min}} and {{max}}: ({{v}})',
+  //name: 'range',
```
Comment (mymattcarroll) [inline src/lib/forms-schema/common.ts:48]: Can we move please.

File: src/lib/forms-schema/common.ts [hunk 2]
```diff
@@ -43,39 +43,48 @@ export const SEARCH_OPTION_TYPE = 'SEARCH'
...
+          assert: joi.number(),
+        },
+      ],
+      validate(value, helpers, args, options) {
+        console.log({
+          value,
+          helpers,
+          args,
+          options,
+        })
```
Comment (mymattcarroll) [inline src/lib/forms-schema/common.ts:76]: Can we remove this please or add some content to the log before the object is logged.

File: src/lib/forms-schema/elements/RepeatableSetElement.ts [hunk 3]
```diff
@@ -8,10 +8,11 @@ import {
   readOnly,
   conditionallyShowSchemas,
 } from '../property-schemas'
-import elementSchema from '../element-schema'
+// import elementSchema from '../element-schema'
```
Comment (mymattcarroll) [inline src/lib/forms-schema/elements/RepeatableSetElement.ts:11]: Remove please

File: tests/forms-schema.test.ts [hunk 4]
```diff
@@ -1,825 +1,820 @@
...
+          element.type === 'form' ||
+          element.type === 'infoPage' ||
+          element.type === 'image' ||
+          element.type === 'html' ||
+          element.type === 'heading'
+        ) {
+          // @ts-expect-error ???
+          expect(element.readOnly).toBeUndefined()
+        } else {
+          // @ts-expect-error ???
```
Comment (mymattcarroll) [inline tests/forms-schema.test.ts:443]: Can we put these comments back in instead of ???

File: tests/forms-schema.test.ts [hunk 5]
```diff
@@ -1,825 +1,820 @@
...
         element.type === 'image' ||
         element.type === 'html' ||
         element.type === 'heading'
       ) {
-        // @ts-expect-error "readyOnly" should not be here, hence the check
+        // @ts-expect-error ???
         expect(element.readOnly).toBeUndefined()
       } else {
-        // @ts-expect-error "readyOnly" should be here, hence the check
+        // @ts-expect-error ???
```
Comment (mymattcarroll) [inline tests/forms-schema.test.ts:809]: Same here

File: tests/forms-schema.test.ts [hunk 6]
```diff
@@ -1223,21 +1082,144 @@ test('should error if "toDate" is greater than "fromDate"', () => {
...
+  })
+
+  expect(error?.message).toContain(
+    '"Form Element - Maximum Number" must be an integer',
+  )
+})
+
+// Looks like this is impossible with the current Joi version
+// https://github.com/hapijs/joi/issues/1685
+test.skip('should throw error if maxNumber is not provided for number element with isSlider as true', () => {
```
Comment (mymattcarroll) [inline tests/forms-schema.test.ts:1165]: Can this be achieved now?

Comment (mymattcarroll) [reply tests/forms-schema.test.ts:1165]: Added https://blinkmobile.atlassian.net/browse/ON-12471  sub task

File: tests/forms-schema.test.ts (hunk 5 above)
Comment (Zaxist) [reply tests/forms-schema.test.ts:809]: lol, yeh

**Pull request merged**